pkg/json_test.go

```diff
@@ -490,3 +491,99 @@ func TestJSONOutput(t *testing.T) {
 		})
 	}
 }
+
+func TestScorecardResult_AsJSON(t *testing.T) {
+	t.Parallel()
+	type fields struct {
+		Repo       RepoInfo
+		Date       time.Time
+		Scorecard  ScorecardInfo
+		Checks     []checker.CheckResult
+		RawResults checker.RawResults
+		Metadata   []string
+	}
+	type args struct { //nolint:govet
+		showDetails bool
+		logLevel    log.Level
+	}
+	tests := []struct { //nolint:govet
+		name       string
+		fields     fields
+		args       args
+		wantWriter string
+		wantErr    bool
+	}{
+		{
+			name: "default",
+			fields: fields{
+				Repo: RepoInfo{
+					Name:      "repo",
+					CommitSHA: "123456789012345678",
+				},
+			},
+			wantWriter: `{"Repo":"repo","Date":"0001-01-01","Checks":null,"Metadata":null}`,
```

Comment:
we should probably include a date in the fields above, if its included in the output